samples/hello/hello_treeview.rb

```diff
@@ -0,0 +1,43 @@
+require 'glimmer-dsl-tk'
+
+class TreeData
+  attr_accessor :selection, :selection_options
+
+  def initialize
+    @selection_options = [
+      'Item 1',
+      'Item 2',
+      'Tree 1' => [
+        'Subitem 1',
+        'Subitem 2',
+        'Subtree 1' => [
+          'And so on...'
+        ]
+      ]
+    ]
+  end
+end
+
+class HelloTreeview
+  include Glimmer
+
+  def launch
+    tree_data = TreeData.new
+
+    root {
+      center_within_screen
+      width 600
+      height 600
+
+      title 'Hello, Treeview!'
+      treeview {
+        grid sticky: 'nsew', row_weight: 1, column_weight: 1
+        show 'tree'
+        selectmode 'browse'
+        selection <=> [tree_data, :selection]
```

Comment:
I tested this, and I appreciate that you had the selection data-binding infer options automatically just like we do with lists. 

When I traced the selection value on the model though, I was expecting to see only a single string representing the node that was selected. Instead, I got an array, and it included a hash and another array the deeper I got.

I added this code to the model to do the printout:

```ruby  
  def selection=(value)
    @selection=value
    pd @selection
  end
```

`pd` (https://github.com/AndyObtiva/puts_debuggerer) is similar to `puts`, but prints more information like the file name and line number, and uses `ap` (awesome print), so I like to use it for debugging when I am not using a real debugger.

Printout:
```
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => []
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] "Item 1"
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] "Item 2"
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] "Tree 1"
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] {
        "Tree 1" => [
            [0] "Subitem 1"
        ]
    }
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] {
        "Tree 1" => [
            [0] "Subitem 2"
        ]
    }
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] {
        "Tree 1" => [
            [0] "Subtree 1"
        ]
    }
]
[PD] samples/hello/hello_treeview.rb:22
   > pd @selection
  => [
    [0] {
        "Tree 1" => [
            [0] {
                "Subtree 1" => [
                    [0] "And so on..."
                ]
            }
        ]
    }
]
```

For the basic top-level nodes, it seemed strange to return an array like `['item1']`. I would have expected to just get `item1`. For the deeper levels, I would have also expected to get a single string too.

I am curious, is there a reason why you implemented tree selection data-binding that way? Is that how you use the data in your applications?